src/sentry/api/endpoints/organization_member/team_details.py

```diff
@@ -165,21 +177,102 @@ def get(
             serialize(omt, request.user, OrganizationMemberTeamDetailsSerializer()), status=200
         )
 
+    @extend_schema(
+        operation_id="Add an Organization Member to a Team",
+        parameters=[
+            GlobalParams.ORG_SLUG,
+            GlobalParams.member_id("The ID of the organization member to add to the team"),
+            GlobalParams.TEAM_SLUG,
+        ],
+        request=None,
+        responses={
+            201: BaseTeamSerializer,
+            202: RESPONSE_ACCEPTED,
+            204: RESPONSE_NO_CONTENT,
+            401: RESPONSE_UNAUTHORIZED,
+            403: OpenApiResponse(
+                description="This team is managed through your organization's identity provider"
+            ),
+            404: RESPONSE_NOT_FOUND,
+        },
+        examples=TeamExamples.ADD_TO_TEAM,
+    )
     def post(
         self,
         request: Request,
         organization: Organization,
         member: OrganizationMember,
         team_slug: str,
     ) -> Response:
-        """
-        Join, request access to or add a member to a team.
-        If the user needs permission to join the team, an access request will
-        be generated and the returned status code will be 202.
-        If the user is already a member of the team, this will simply return
-        a 204.
-        If the team is provisioned through an identity provider, then the user
-        cannot join or request to join the team through Sentry.
+        # NOTE: Required to use HTML for table b/c this markdown version doesn't support colspan.
+        r"""
+        This request can return various success codes depending on the context of the team:
+        - **`201`**: The member has been successfully added.
+        - **`202`**: The member needs permission to join the team and an access request
+        has been generated.
+        - **`204`**: The member is already on the team.
+
+        If the team is provisioned through an identity provider, then the member cannot join the
+        team through Sentry.
+
+        Note that the permission scopes also vary depending on the organization setting `"Open
+        Membership"` and the type of authorization token. The following table outlines the accepted
+        scopes.
+        <table style="width: 100%;">
```

Comment:
Do you think this can be used by other APIs as well? If so can you create a util method that would generate the html from a python map or something? Ideally can this be part of extend_schema?

Reply:
I don't think this is extensible to other APIs for now, probably just the ones in this file b/c they all involve Team Admin